src/core/batch/batch.ts

```diff
@@ -27,25 +27,26 @@ interface BatchFileData {
   hash: HexString;
 }
 
-type SignedAnnouncementIterable<T extends AnnouncementType> = AsyncOrSyncIterable<
-  AnnouncementWithSignature<TypedAnnouncement<T>>
->;
+type SignedAnnouncementIterable<T extends AnnouncementType> = AsyncOrSyncIterable<AnnouncementWithSignature<T>>;
 
-type ParquetRecord = Record<string, Buffer | Uint8Array | BigInt | number>;
+type ParquetRecord = Record<string, Buffer | Uint8Array | bigint | number>;
 
 const parseAnnouncement = <T extends SignedAnnouncement>(record: ParquetRecord): T => {
-  const schema = getSchemaFor(Number(record.announcementType));
-  const announcement: Record<string, string | number | BigInt> = {};
-
-  for (const key in schema) {
-    if (schema[key].type === "BYTE_ARRAY") {
-      announcement[key] = record[key].toString();
-    } else if (typeof record[key] === "number") {
-      announcement[key] = Number(record[key]);
-    } else if (typeof record[key] === "bigint") {
-      announcement[key] = BigInt(record[key].toString());
-    }
-  }
+  const schemas = getSchemaFor(record.announcementType as AnnouncementType);
+
+  const announcement = Object.entries(schemas).reduce<Record<string, string | number | bigint>>(
```

Comment:
The types really didn't like working with schema and announcement in this way, so I switched over to a reduce setup.